packages/shared/src/modules/points-of-interest/libs/validation-schemas/points-of-interest-search-query.validation-schema.ts

```diff
@@ -0,0 +1,48 @@
+import { z } from "zod";
+
+import { parseToFloat } from "../../../../libs/helpers/helpers.js";
+import {
+	PointsOfInterestValidationMessage,
+	PointsOfInterestValidationRule,
+} from "../enums/enums.js";
+import {
+	latitudeSchema,
+	longitudeSchema,
+} from "./longitude-latitude.validation-schema.js";
+
+const pointsOfInterestSearchQuery = z
+	.object({
+		latitude: latitudeSchema.optional(),
+		longitude: longitudeSchema.optional(),
+		radius: z
+			.string()
+			.transform(parseToFloat)
+			.pipe(
+				z
+					.number()
+					.min(
+						PointsOfInterestValidationRule.RADIUS_MIN_KM,
+						PointsOfInterestValidationMessage.RADIUS_MIN_KM,
+					)
+					.max(
+						PointsOfInterestValidationRule.RADIUS_MAX_KM,
+						PointsOfInterestValidationMessage.RADIUS_MAX_KM,
+					),
+			)
+			.optional(),
+	})
+	.refine(
+		({ latitude, longitude, radius }) =>
+			!radius || (Boolean(latitude) && Boolean(longitude)),
+		{
+			message: PointsOfInterestValidationMessage.COORDS_REQUIRED_WITH_RADIUS,
+		},
+	)
+	.refine(
+		({ latitude, longitude }) => Boolean(latitude) === Boolean(longitude),
```

Comment:
from logical perspective it should be and, not equality
```suggestion
		({ latitude, longitude }) => Boolean(latitude) && Boolean(longitude),
```

Reply:
No, after this changes I can not sent request without query.

Reply:
oh i see, I was thinking that we need both options present